projects/mci/src/mci_output.c

```diff
@@ -0,0 +1,81 @@
+#include "mci_output.h"
+
+#include <string.h>
+
+#include "motor_can.h"
+#include "motor_controller.h"
+#include "wavesculptor.h"
+
+#include "exported_enums.h"
+#include "generic_can.h"
+#include "pedal_rx.h"
+#include "soft_timer.h"
+#include "status.h"
+
+static float s_velocity_lookup[] = {
+  [EE_DRIVE_OUTPUT_OFF] = 0.0f,
+  [EE_DRIVE_OUTPUT_DRIVE] = WAVESCULPTOR_FORWARD_VELOCITY,
+  [EE_DRIVE_OUTPUT_REVERSE] = WAVESCULPTOR_REVERSE_VELOCITY,
+};
+
+static float s_max_regen_in_throttle = 0.0f;
+static float s_regen_threshold = 0.0f;
+
+// Basic implementation of throttle/brake maps
+static float prv_brake_to_regen_map(float brake_value) {
+  return brake_value / PEDAL_RX_MAX_PEDAL_VALUE;
+}
+
+static float prv_throttle_to_regen_map(float throttle_value) {
+  return 0.0f;
+}
+
+static float prv_throttle_to_accel_map(float throttle_value) {
+  return throttle_value / PEDAL_RX_MAX_PEDAL_VALUE;
+}
+
+static void prv_send_wavesculptor_message(MotorControllerStorage *storage,
+                                          MotorControllerId motor_controlller_id,
+                                          MotorCanDriveCommand command) {
+  GenericCanMsg msg = {
+    .id = motor_controlller_id,
+    .dlc = MOTOR_CAN_DRIVE_COMMAND_LENGTH,
+    .extended = MOTOR_CAN_DRIVE_COMMAND_IS_EXTENDED,
+  };
+
+  uint8_t data[MOTOR_CAN_DRIVE_COMMAND_LENGTH] = { 0 };
+  motor_can_drive_command_pack(data, &command, sizeof(data));
+  memcpy(&msg.data, data, sizeof(data));
+  generic_can_tx(storage->motor_can, &msg);
+}
+
+static void prv_handle_drive(SoftTimerId timer_id, void *context) {
+  MotorControllerStorage *storage = context;
+  PedalValues pedal_values = pedal_rx_get_pedal_values(&storage->pedal_storage);
+  MotorCanDriveCommand drive_command = { 0 };
+  // TODO(SOFT-122): Make sure test ensures that maps are continues
+  if (storage->drive_state == EE_DRIVE_OUTPUT_OFF) {
```

Comment:
instead of accessing it from storage directly,  access it from the drive_fsm. I left Jess a PR to change it.